alns/stop/tests/test_no_improvement.py

```diff
@@ -7,29 +7,42 @@
 from alns.stop import NoImprovement
 
 
-@pytest.mark.parametrize("n_iterations", [-10, -100, -1000])
-def test_raise_negative_parameters(n_iterations: int):
+@pytest.mark.parametrize("max_iterations", [-10, -100, -1000])
+def test_raise_negative_parameters(max_iterations: int):
     """
-    n_iterations cannot be negative.
+    max_iterations cannot be negative.
     """
     with assert_raises(ValueError):
-        NoImprovement(n_iterations)
+        NoImprovement(max_iterations)
 
 
-@pytest.mark.parametrize("n_iterations", [0, 10, 100, 1000])
-def test_does_not_raise(n_iterations: int):
+@pytest.mark.parametrize("max_iterations", [0, 10, 100, 1000])
+def test_does_not_raise(max_iterations: int):
     """
     Non-negative integers should not raise.
     """
-    NoImprovement(n_iterations)
+    NoImprovement(max_iterations)
 
 
-def test_n_iterations():
+def test_max_iterations():
     """
-    Test if the n_iterations parameter is correctly set.
+    Test if the max_iterations parameter is correctly set.
     """
     stop = NoImprovement(3)
-    assert stop.n_iterations == 3
+    assert_equal(stop.max_iterations, 3)
+
+
+def test_zero_max_iterations():
+    """
+    Test if setting max_iterations to zero stops when a non-improving
+    best solution has been found.
+    """
+    stop = NoImprovement(0)
+    rnd = RandomState()
+
+    assert_(not stop(rnd, One(), Zero()))
+    assert_(not stop(rnd, Zero(), Zero()))
+    assert_(stop(rnd, Zero(), Zero()))
```

Comment:
Issue: this feels wrong, in a sense. No improving solution can be found in no iterations. So for zero iterations we should always stop.

I suggest we fix this by implementing `__call__` slightly differently.